cliconfig/config_test.go

```diff
@@ -11,28 +11,37 @@ import (
 	"github.com/docker/docker/pkg/homedir"
 )
 
-func TestMissingFile(t *testing.T) {
+func TestEmptyConfigDir(t *testing.T) {
 	tmpHome, _ := ioutil.TempDir("", "config-test")
+	defer os.RemoveAll(tmpHome)
 
-	config, err := Load(tmpHome)
+	SetConfigDir(tmpHome)
+
+	config, err := Load("")
 	if err != nil {
-		t.Fatalf("Failed loading on missing file: %q", err)
+		t.Fatalf("Failed loading on empty config dir: %q", err)
 	}
```

Comment:
Since you're using SetConfigDir() + Load("") you may want to add a call to config.Filename() to make sure its tmpHome+config.json.  Just to make sure Load("") uses the configDir we set.